iOS/Chameleon/source/Chameleon/ApiDefinition.cs

```diff
@@ -16,524 +16,591 @@
+        [Static]
+        [Export("colorWithAverageColorFromImage:withAlpha:")]
+        UIColor GetImageAverageColor(UIImage image, nfloat alpha);
+
+        // +(UIColor *)colorWithComplementaryFlatColorOf:(UIColor *)color;
+        [Static]
+        [Export("colorWithComplementaryFlatColorOf:")]
+        UIColor ColorWithComplementaryFlatColorOf(UIColor color);
+
+        // +(UIColor *)colorWithComplementaryFlatColorOf:(UIColor *)color withAlpha:(CGFloat)alpha;
+        [Static]
+        [Export("colorWithComplementaryFlatColorOf:withAlpha:")]
+        UIColor ColorWithComplementaryFlatColorOf(UIColor color, nfloat alpha);
+
+        // +(UIColor *)colorWithContrastingBlackOrWhiteColorOn:(UIColor *)backgroundColor isFlat:(BOOL)flat;
+        [Static]
+        [Export("colorWithContrastingBlackOrWhiteColorOn:isFlat:")]
+        UIColor ColorWithContrastingBlackOrWhiteColorOn(UIColor backgroundColor, bool flat);
+
+        // +(UIColor *)colorWithContrastingBlackOrWhiteColorOn:(UIColor *)backgroundColor isFlat:(BOOL)flat alpha:(CGFloat)alpha;
+        [Static]
+        [Export("colorWithContrastingBlackOrWhiteColorOn:isFlat:alpha:")]
+        UIColor ColorWithContrastingBlackOrWhiteColorOn(UIColor backgroundColor, bool flat, nfloat alpha);
+
+        // +(UIColor *)colorWithGradientStyle:(UIGradientStyle)gradientStyle withFrame:(CGRect)frame andColors:(NSArray *)colors;
+        [Static]
+        [Export("colorWithGradientStyle:withFrame:andColors:")]
+
+        UIColor ColorWithGradientStyle(UIGradientStyle gradientStyle, CGRect frame, NSObject[] colors);
+
+        // +(UIColor *)colorWithHexString:(NSString *)string;
+        [Static]
+        [Export("colorWithHexString:")]
+        UIColor ColorWithHexString(string @string);
+
+        // +(UIColor *)colorWithHexString:(NSString *)string withAlpha:(CGFloat)alpha;
+        [Static]
+        [Export("colorWithHexString:withAlpha:")]
+        UIColor ColorWithHexString(string @string, nfloat alpha);
+
+        // -(UIColor *)flatten;
+        [Export("flatten")]
+
+        UIColor Flatten();
+
+        // -(UIColor *)darkenByPercentage:(CGFloat)percentage;
+        [Export("darkenByPercentage:")]
+        UIColor DarkenByPercentage(nfloat percentage);
+
+        // -(NSString *)hexValue;
+        [Export("hexValue")]
+
+        string HexValue();
+
+        // -(UIColor *)lightenByPercentage:(CGFloat)percentage;
+        [Export("lightenByPercentage:")]
+        UIColor LightenByPercentage(nfloat percentage);
+
+        // +(UIColor *)colorWithFlatVersionOf:(UIColor *)color __attribute__((deprecated(" Use -flatten: instead (First deprecated in Chameleon 2.0).")));
+        [Static]
+        [Export("colorWithFlatVersionOf:")]
+        UIColor ColorWithFlatVersionOf(UIColor color);
+    }
+
+    // @interface Chameleon (NSArray)
+    [Static]
+    [BaseType(typeof(NSObject), Name = "NSArray")]
+    interface ChameleonColorArray
```

Comment:
Binding static stuff inside categories is a bad idea due the only way to access them is through `(null as Foo).FooMethod ()`. What I recommend is to leave the class as before ([1][1] and [2][2]) then, remove the `BaseType` due is not necessary.

[1]: https://github.com/adlair/XamarinComponents/blob/4bfb9e9ff5d2a42b827c3ce80acb523ff3c63eec/iOS/Chameleon/source/Chameleon/ApiDefinition.cs#L396
[2]: https://github.com/adlair/XamarinComponents/blob/4bfb9e9ff5d2a42b827c3ce80acb523ff3c63eec/iOS/Chameleon/source/Chameleon/Additions.cs#L20